### PR TITLE
Jvenstad/deploy to dev jobs

### PR DESCRIPTION
**Pull request by jonmv**
@freva please review.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobProfile.java [hunk 1]
```diff
@@ -64,6 +68,7 @@ public static JobProfile of(JobType type) {
             case test: return systemTest;
             case staging: return stagingTest;
             case prod: return production;
+            case dev: return development;
```
Comment (freva) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobProfile.java]: `case perf`?

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/deployment/ApplicationStore.java [hunk 2]
```diff
@@ -37,4 +37,10 @@
     /** Removes all tester packages for the given tester. */
     void removeAll(TesterId tester);
 
+    /** Stores the given application package as the development package for the given application. */
+    void putDev(ApplicationId application, byte[] applicationPackage);
```
Comment (freva) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/deployment/ApplicationStore.java]: Will there be a separate method for `perf` or maybe rename this more generally and add extra arg?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobProfile.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobProfile.java]: There are no jobs for perf yet. Should we add those right away?

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/deployment/ApplicationStore.java (hunk 2 above)
Comment (jonmv) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/deployment/ApplicationStore.java]: Both are development things, though ...

Comment (jonmv) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/deployment/ApplicationStore.java]: I would like to keep it just one thing. Simpler, less stuff :)

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobProfile.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobProfile.java]: Guess I can do that.

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/deployment/ApplicationStore.java (hunk 2 above)
Comment (freva) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/deployment/ApplicationStore.java]: Sure, I'm just thinking about the implementation of this method, so we don't override the `dev` package when someone deploys to `perf`...

**Review by freva — APPROVED** [review]
Great! :clap:

**Pull request merged**